posts/inside-rust/2024-06-19-this-development-cycle-in-cargo-1.80.md

```diff
@@ -0,0 +1,517 @@
+---
+layout: post
+title: "This Development-cycle in Cargo: 1.80"
+author: Ed Page
+team: The Cargo Team <https://www.rust-lang.org/governance/teams/dev-tools#cargo>
+---
+
+# This Development-cycle in Cargo: 1.80
+
+This is a summary of what has been happening around Cargo development for the last 6 weeks which is approximately the merge window for Rust 1.80.
+
+<!-- time period: 2024-05-03 through 2024-06-13 -->
+
+- [Plugin of the cycle](#plugin-of-the-cycle)
+- [Implementation](#implementation)
+  - [check-cfg](#check-cfg)
+  - [User-controlled cargo diagnostics](#user-controlled-cargo-diagnostics)
+  - [`-Ztrim-paths`](#-ztrim-paths)
+  - [MSRV-aware Cargo](#msrv-aware-cargo)
+  - [Removing implicit features](#removing-implicit-features)
+  - [Normalizing published manifest files](#normalizing-published-manifest-files)
+  - [Merging `cargo upgrade` into `cargo update`](#merging-cargo-upgrade-into-cargo-update)
+  - [`.crate` provenance](#crate-provenance)
+  - [`cargo publish --workspace`](#cargo-publish---workspace)
+  - [Snapshot testing](#snapshot-testing)
+- [Design discussions](#design-discussions)
+  - [RFC triage](#rfc-triage)
+  - [Custom test harnesses and `panic = "abort"`](#custom-test-harnesses-and-panic--abort)
+  - [Short-hand manifest syntaxes](#short-hand-manifest-syntaxes)
+  - [Leaky abstractions of rustc](#leaky-abstractions-of-rustc)
+- [Misc](#misc)
+- [Focus areas without progress](#focus-areas-without-progress)
+
+## Plugin of the cycle
+
+Cargo can't be everything to everyone,
+if for no other reason than the compatibility guarantees it must uphold.
+Plugins play an important part of the Cargo ecosystem and we want to celebrate them.
+
+Our plugin for this cycle is [cargo-expand](https://crates.io/crates/cargo-expand) a more convenient and easier to remember wrapper around `cargo rustc` for seeing all macros expanded within a crate.
+For macro authors, this is a big debugging help.
+For macro users, this can help in breaking the opacity of what a macro invocation does.
+
+Thanks to [LukeMathWalker](https://github.com/LukeMathWalker) for the suggestion!
+
+[Please submit your suggestions for the next post.](https://rust-lang.zulipchat.com/#narrow/stream/246057-t-cargo/topic/Plugin.20of.20the.20Dev.20Cycle/near/420703211)
+
+## Implementation
+
+##### check-cfg
```

Comment:
Could use a tone check on this section because this was a fairly sensitive topic and want to make sure people understand we are sensitive to their concerns.